internal/graph/check_test.go

```diff
@@ -72,3 +72,50 @@ func TestResolveCheckDeterministic(t *testing.T) {
 	require.ErrorIs(t, err, ErrResolutionDepthExceeded)
 	require.Nil(t, resp)
 }
+
+func TestCheckWithOneConcurrentGoroutineCausesNoDeadlock(t *testing.T) {
```

Comment:
Please add this to the test suite in `assets/tests` instead of here. That way we get long-term coverage regardless of the underlying implementation.

Reply:
`server.Check` won't necessarily always delegate to the `graph.LocalChecker` implementation. My point is that we want to make sure that the Check implementation itself doesn't have a deadlock, and so we should test the top-level function as well. If we were to change the implementation of `server.Check` and if it didn't use `graph.LocalChecker`, then we would potentially introduce the problem with some other code.

Reply:
Hm but this test is doing `NewLocalChecker(ds, 1)`

I can't set that parameter through the server settings

Reply:
Ok, this is not a blocker. But we should strive to have functional tests that cover the same behavioral cases as deeply internal tests.

Reply:
I agree, but that doesn't apply here, i'm adding a unit test that tests the goroutine management